### PR TITLE
ISPN-11273 Local Indexes out of sync after segment ownership lost during state transfer

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.redhat.com/browse/ISPN-11273

~~Marking as preview because it depends on https://github.com/infinispan/infinispan/pull/7792~~

### CODE REVIEW

**Comment by anistor** [comment]
I'm eyeballing :eyes: this one now but it needs rebase @gustavonalle

**Comment by gustavonalle** [comment]
Rebased

File: query/src/main/java/org/infinispan/query/impl/IndexInspector.java [hunk 1]
```diff
@@ -62,7 +65,15 @@ public boolean hasSharedIndex(Class<?> entity) {
       IndexManager indexManager = getIndexManagerPerClass().get(entity);
       if (indexManager == null) return false;
       return indexManager instanceof InfinispanIndexManager ||
-            indexManager.getClass().getName().equals("org.hibernate.search.elasticsearch.impl.ElasticsearchIndexManager");
+            indexManager.getClass().getName().equals(ELASTICSEARCH_INDEX_MANAGER);
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/impl/IndexInspector.java]: Here it should just call ```hasSharedIndex```

Comment (anistor) [reply query/src/main/java/org/infinispan/query/impl/IndexInspector.java]: Looks good. I'll fix the minor thing myself and merge it @gustavonalle !

**Comment by anistor** [comment]
Merged. Thanks @gustavonalle !

**Pull request closed without merging**